### PR TITLE
Fail when invalid values are provided to rule arguments.

### PR DESCRIPTION
**Pull request by askreet**
Inspired by #350.

### CODE REVIEW

**Comment by askreet** [comment]
Now properly tells me I am dumb when using arguments incorrectly:

```
526> yabai -m rule --add title="^Opening" manage=false label=rule1
unknown value 'false' for argument 'manage'
```

File: src/message.c [hunk 1]
```diff
@@ -1764,30 +1769,45 @@ static void handle_domain_rule(FILE *rsp, struct token domain, char *message)
                     rule->manage = RULE_PROP_ON;
                 } else if (string_equals(value, ARGUMENT_COMMON_VAL_OFF)) {
                     rule->manage = RULE_PROP_OFF;
+                } else {
+                    bad_rule_argument_value(rsp, key, value);
+                    return;
```
Comment (askreet) [inline src/message.c]: I noticed there is a mix of having an internal value in the object saying it's OK (for regex in many cases) and early return on failure. Not sure which pattern is preferred, can update this to match other pattern, just let me know.

**Comment by koekeishiya** [comment]
I appreciate the effort, but as you mentioned there is actually not a clear way to properly deal with errors regarding invalid arguments as of right now. 

Usually I'd agree with something similar to what you did in this PR, but I think I want to rework how rules (and signals) are parsed into something that can validate the value given to key-value pairs, and also whether the key-value pair is required or optional.

**Comment by koekeishiya** [comment]
Didn't bother too much with what I wrote in my previous comment, but solved it in a way I'm fairly ok with for now.

**Pull request closed without merging**